### PR TITLE
Incorporate Slack discussions with borkdude into README

### PR DESCRIPTION
**Pull request by lread**
- Under "Usage" section, the reader is introduced to bindings. The println
binding example might be interpreted as how one should use println with sci.
Now mentioning that this simply an illustrative example.
- Under "Vars" section, add a few of borkdude's Slack chat words on rationale
for vars.
- Under "Stdout and stdin" section, added an example of how an included Clojure
function could map to sci's out.
- Add a "Positional metadata" section - this confused me when I was
experimenting with rewrite-cljc under sci.

### CODE REVIEW

**Comment by lread** [comment]
My best shot at distilling our discussions on Slack without getting too wordy. Lemme know if you want any changes.

**Comment by borkdude** [comment]
Thanks a lot!
